### ISSUE TITLE
Default showTerminalOnLaunch to false

### ISSUE THREAD

**Issue by janaagaard75** [jest-community/vscode-jest#882]
`vscode-jest version`: 4.6.0

Really great that the Jest extension now has the `showTerminalOnLaunch`, but I wish this would default to false and not true. I fear a future where a lot of extensions open the terminal at launch making it impossible to avoid this unless all of them have a showTerminalOnLaunch setting. Since Jest is a pretty popular extension, I think they should set the example of not showing the terminal on launch by default.

**Comment by connectdotz** [comment]
the purpose of showing the terminal is for test-run discovery. I agree it could be a subjective preference, thus the setting. 

Maybe because I always have the terminal panel opened, so wasn't sure what is the negative UI experience you guys were seeing, can you elaborate a bit to help us understand the core issue? Is it because you don't usually have the terminal panel visible at launch?

**Comment by janaagaard75** [comment]
No, I do not open the terminal at launch. Our app is divided into three projects, I use a separate shell window to launch them, and then open the appropriate project in VSCode, depending what I have to work on. Doing it so allows me to switch between projects without having to restart the app. I do use VSCode's terminal, but I close it again when I am done, because I want to save screen real estate. For me the test status in the status bar is plenty enough.

Also: What should happen if other extensions started doing the same? If they all open a terminal, wouldn't they start competing for being that one that gets to one that is visible by default?

**Comment by connectdotz** [comment]
> Also: What should happen if other extensions started doing the same? If they all open a terminal, wouldn't they start competing for being that one that gets to one that is visible by default?

They might. However, I think the right question is not about "competing with other extensions", but to decide when we must show users the information. Right now, it is unconditional - the terminal is shown on launch; I think we could perhaps make it by default to only show the terminal when there is an "error"... In such a case the test status might be reflected properly so important for users to see more information, similar to how we show OUTPUT panel right now.

thoughts?

**Comment by janaagaard75** [comment]
> we could perhaps make it by default to only show the terminal when there is an "error".

I think that would be a vast improvement. I would probably turn on that feature, so for me, this might as well be the default behavior.

That would mean allowing strings as settings, right? Perhaps with `true` and `false` also included as valid settings for a while, so that we don't break backwards compatibility right away.

    "jest.showTerminalOnLaunch": "always" | "onErrors" | "never"
